docs/COMMIT.md

```diff
@@ -21,8 +21,15 @@ Format Code
   | xargs clang-format-11 -i
   ```
  
-To format only lines you added with `git add` you can call `git clang-format-11` before you create a commit.
-Take care that un-staged changes will not be formatted.
+Instead of using the bash command above you can use *Git* together with *ClangFormat* to format your patched code only. 
+Before applying this command, you must extend your local git configuration **once** with all file endings used in *PIConGPU*:
+
+```
+git config --local clangFormat.extensions def,h,cpp,cu,hpp,tpp,kernel,loader,param,unitless
+```
+
+For only formatting lines you added using `git add`, call `git clang-format-11` before you create a commit.
```

Comment:
I think this is not correct. Perhaps should be `git clang-format` ? At least only this way works on the dev system.

Reply:
I kindly disagree, `git clang-format-11` worked for me on my own system.

Reply:
Okay, so then it's system-specific.

Reply:
git clang-format-11 is using definitiv clangFormat version11. if you only use 'git clang-format' it can be any version depending on your system or what you defined in the git config.

Reply:
Yet on our dev system there is clang format 11 installed, and only `clang-format` works (and is version 11), but not `clang-format-11`. I am not sure, maybe this is some issue on the system.

Reply:
It depends on how the module is provided. On our dev system, Clang11 is compiled via spack. I would like to keep the name `clang-format-11` to be clear which version is used. If an OS does not have `clang-format-11` and `clang-format` is available I think the developer will manage it. 
I do not like to describe edge cases for each OS/distribution.

Reply:
If you like I can add a warning box in the `rst` file, I would not add it to the `md` file.

```
The binary for `ClangFormat` is on some operating system called `clang-format`. If so please check that `clang-format --version` return version `11.X.X`.
```

Reply:
Okay, that seems good to me

Reply:
I added the warning to the RST file